src/Compilers/Core/Portable/Binding/BindingDiagnosticBag.cs

```diff
@@ -252,10 +258,10 @@ internal bool AddDiagnostics(Location location, CompoundUseSiteInfo<TAssemblySym
             return false;
         }
 
-        internal bool Add(Location location, CompoundUseSiteInfo<TAssemblySymbol> useSiteInfo)
+        internal bool Add<TData>(TData data, CompoundUseSiteInfo<TAssemblySymbol> useSiteInfo, Func<TData, Location> getLocation)
```

Comment:
>internal bool Add<TData>(TData data, CompoundUseSiteInfo<TAssemblySymbol> useSiteInfo, Func<TData, Location> getLocation) [](http://example.com/codeflow?start=8&length=121)

It might be worth having an overload of this helper that takes ```Location``` as the original method used to do. In order to avoid allocating delegates for those consumers.

Reply:
overall, that would save 3-4 allocations total, at the cose of needing a path through all of this distinct from the current uniform generic path.  Whereas, this PR saves many thousands or millions of allocations for teh common case in medium to large compilations.  I don't see the value adding the overload to save such a small amount of allocations, especially with the maintenance burden of having to keep the overloads in sync.